internal/pkg/envconfig/envconfig.go

```diff
@@ -11,18 +11,23 @@ import (
 	"os"
 )
 
-// TODO:
-//
-// Vars like 'SINGULARITY_*' should be declared as const here
-//
+const (
+	DisableCache = "SINGULARITY_DISABLE_CACHE"
+	CacheDir     = "SINGULARITY_CACHEDIR"
+)
 
 // NoCache will return true if 'SINGULARITY_DISABLE_CACHE' is set.
 func NoCache() bool {
-	env := os.Getenv("SINGULARITY_NO_CACHE")
-	return env != ""
+	env, ok := os.LookupEnv(DisableCache)
+
+	if !ok || env == "" {
```

Comment:
```
if !ok {
    return false
}
```

Env being "" should be fine. As long as variable is set.

Reply:
Wait, if `SINGULARITY_DISABLE_CACHE` is set to `""`, then it *should* use cache? Right now it works like:

```bash
$ sudo SINGULARITY_DISABLE_CACHE="" singularity build -F test.sif test.def 
INFO:    Starting build...
# uses the cache image
/root/.singularity/cache/library/sha256.63259fd0a2acb88bb652702c08c1460b071df51149ff85dc88db5034532a14a0/alpine_latest.sif
INFO:    Creating SIF file...
INFO:    Build complete: test.sif

$ sudo SINGULARITY_DISABLE_CACHE="1" singularity build -F test.sif test.def 
INFO:    Starting build...
# downloads a new image
INFO:    Downloading library image to tmp cache: /tmp/single_use_cache_520818175
INFO:    Creating SIF file...
INFO:    Build complete: test.sif
```

Reply:
This comes down to... Do you view the variable being set as enough to disable the cache? Even if it's a empty value? 

The `SINGULARITY_DISABLE_CACHE` variable is set, even though it's an empty string. You're setting the environment variable that disables the use of cache.

Reply:
Imo, if `SINGULARITY_DISABLE_CACHE` is set to `""`, then thats the same as a unset variable. So if `SINGULARITY_DISABLE_CACHE` have a value, then singularity wont use the cache. That makes sense to me, imo.